presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -607,6 +607,27 @@ core::PlanNodePtr VeloxQueryPlanConverterBase::toVeloxQueryPlan(
             left->outputType()));
   }
 
+  // For ScanFilter and ScanFilterProject, the planner sometimes put the
+  // remaining filter in a FilterNode after the TableScan.  We need to put it
+  // back to TableScan so that Velox can leverage it to do stripe level
+  // skipping.  Otherwise we only get row level skipping and lose some
+  // optimization opportunity in case of very low selectivity.
+  if (auto tableScan = std::dynamic_pointer_cast<const protocol::TableScanNode>(
+          node->source)) {
+    if (auto* tableLayout = dynamic_cast<protocol::HiveTableLayoutHandle*>(
+            tableScan->table.connectorTableLayout.get())) {
+      auto remainingFilter =
```

Comment:
This code looks a bit strange in that tableLayout->remainingPredicate is a boolean expression evaluating to true, and its reset to the filter predicate instead. So you are suggesting the Presto planner first moves this around and then we want to move it back ?

Reply:
Yes we only reset it when it is a constant true (no-op).  This is also the pattern I observed in query.

Reply:
Yes it should be empty values node if the condition is constant false